code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm

```diff
@@ -547,6 +547,14 @@
 					item_state = "ginvodkaglass"
 					name = "glass of Grey vodka"
 					desc = "A questionable concoction of objects found within maintenance. Tastes just like you'd expect."
+				if(PINTPOINTER)
+					var/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/pintpointer/P = new (get_turf(src))
```

Comment:
The problem with this is that if you mix it while it's in your hand you'll drop it on the floor which is not great.

Also, `get_turf()` is probably a bad choice.

Reply:
I agree that it's not great, but I think it'd be worse to put all the Pintpointer code into `/drinkingglass` and just have it only function when the glass is full of pintpointer.

Reply:
Could at the very least do some checks to see if it's in the hand of a mob and swap it out of it is. Won't be fool proof but it's already an improvement.

Bonus points if you make it a generic proc so more code can do this.

Reply:
I did that but never pushed it
Just... leave it like this for now and I'll actually finish that system after my current PR is done